Automattic/sensei
# Proposal: Switch to using Composer autoloader for Sensei classes

Resolves #6754

Now that we're using Composer's autoloader anyway, we should switch to create a pre-built classmap. These are more performant at runtime because we're not doing a `file_exists` for every loaded Sensei class using the old bundles. During the introduction of a new class during development, you can run `composer install` or just `composer dump-autoload` to regenerate the map with your new class. **We might want to consider adding a post-checkout (or some other git hook) to run this automatically when switching branches.**

## Proposed Changes
* Deprecate Sensei's autoloaders.
* Build a classmap during build time. 

## Testing Instructions

1. Make sure Sensei doesn't explode.
2. Try with known plugins to check compatibility.

## Deprecated Code


* `Sensei_Autoloader_Bundle`
* `Sensei_Autoloader`
* `Sensei_Bootstrap`

## Pre-Merge Checklist
